Wketchum/genie generator

wesketchum opened the pull request:
PR for inclusion of GENIE generation, including storing of GENIE truth information.

Should be included with https://github.com/LDMX-Software/cmake/pull/21.

wesketchum committed:
initial commit of a GenieGenerator. runs one event before it crashes...

tomeichlersmith committed:
need to convert energy units

GENIE uses GeV as its internal energy unit while Geant4 uses MeV, we
simply multiply the energy by the constant representing its unit to do
the conversion so that everyone can see what is happening

wesketchum committed:
fix the physics list, add in random seeds, and ... it might actually be working

wesketchum committed:
remove a cout

wesketchum committed:
some more cleanup, especially on messages and logging; also print out total xsec

wesketchum committed:
add in ability to do abundances in generation

wesketchum committed:
Merge branch 'trunk' of https://github.com/LDMX-Software/SimCore into wketchum/Genie_Generator

wesketchum committed:
little bit of cleanup

wesketchum committed:
fix for event weighting by xsec*abundance

wesketchum committed:
Merge branch 'trunk' of https://github.com/LDMX-Software/SimCore into wketchum/Genie_Generator

wesketchum committed:
add randomization over target thickness, use common seeds for random numbers from G4Random/RandomSeedService

wesketchum committed:
Merge remote-tracking branch 'origin/trunk' into wketchum/Genie_Generator

wesketchum committed:
Merge branch 'trunk' of https://github.com/LDMX-Software/SimCore into wketchum/Genie_Generator

wesketchum committed:
Merge branch 'trunk' into wketchum/Genie_Generator

wesketchum committed:
updates for GTruth to be added and filled

wesketchum committed:
Merge remote-tracking branch 'local_21Sep2023/wketchum/Genie_Generator' into wketchum/Genie_Generator

wesketchum committed:
updates for storing all of the GENIE information needed for reweighting into the event

wesketchum committed:
add ability to simulate across a beam spot size at target

wesketchum committed:
Merge branch 'trunk' of https://github.com/LDMX-Software/SimCore into wketchum/Genie_Generator

wesketchum committed:
Merge remote-tracking branch 'origin/trunk' into wketchum/Genie_Generator

wesketchum committed:
Merge branch 'wketchum/Genie_Generator' of https://github.com/LDMX-Software/SimCore into wketchum/Genie_Generator

EinarElen commented:
For the new simulator-code, I think it is worth knowing that we have two producers that are doing the simulator work. One is the simulator and the other is the so-called resimulator. They are both derived from the SimulatorBase class so we should make sure to put common code (like adding the genie particles, fillGTruth etc) happens in there and that anything that needs to go directly in the simulator also goes into the resimulator.Looking at how the seeding of the Genie generator is set up I think it will work automatically with the ReSimulator :)

tomeichlersmith committed:
add option to filter by run number as well as event number

This is helpful for people like me who have massive samples of
heavily-filtered simulations. The best way to generate these samples is
across many runs but then merge the resulting files. This results in
several runs per file and often an event from one run is aborted by the
simulation while the same event number from a different run is the event
I wish to resimulate.

tomeichlersmith committed:
refactor to move config deducing into python

the translation of different format of runs provided by the user into
the actual data structure used during processing is done within python
now so it can be more transparent in error

EinarElen committed:
Add decay as a process type

EinarElen committed:
Add the default kaon physics settings

EinarElen committed:
Include them in simulator

EinarElen committed:
Add a configuration corresponding to the upKaons branch

EinarElen committed:
Add and register kaonphysics as a new constructor

EinarElen committed:
Enum of branching ratios

EinarElen committed:
Add parameters for both particles

EinarElen committed:
Constructor

EinarElen committed:
Add function to change decay properties of kaons

EinarElen committed:
Construct function for particles

EinarElen committed:
Document parameters

EinarElen committed:
Document upKaons

EinarElen committed:
Thanks clang-format checker!

EinarElen committed:
Remove unused interface

EinarElen committed:
Remove from constructor as well

EinarElen committed:
Format and add neutral kaons

EinarElen committed:
Remove note

EinarElen committed:
Doxygen patch

tomeichlersmith committed:
put check on if an event should be skipped into its own function

this is helpful isolation and makes it easier to document how an event
ends up being re-simulated

EinarElen committed:
Add parameter checking

EinarElen committed:
Correct the check, give more details

EinarElen committed:
Add a repr

EinarElen committed:
Don't miss actually setting the lifetimes...

EinarElen committed:
Add further processes

EinarElen committed:
Set process as primary if parent is 0

EinarElen committed:
Add muon nuclear as an interaction type

EinarElen committed:
Inelastic neutron interactions

EinarElen committed:
Inelastic kaon interactions

EinarElen committed:
Inelastic pion interactions

EinarElen committed:
Same for protons

EinarElen committed:
New primary category

EinarElen committed:
Merge pull request #109 from LDMX-Software/107-increase-the-number-of-processes-in-the-process-map

107 increase the number of processes in the process map

EinarElen committed:
Add helper function to write out details

EinarElen committed:
Add verbosity setting

EinarElen committed:
Update orderings

EinarElen committed:
Use dedicated logging

EinarElen committed:
Comments on why order matters

EinarElen committed:
Do logging at info level

We are already gated by the verbosity setting

EinarElen committed:
Prettier formatting for logs

EinarElen committed:
Correct pi-inelastic names

tomeichlersmith committed:
remove EventDef header

tomeichlersmith committed:
remove generator that is not compiling and unused

we /could/ rework this generator to use the new EventFile syntax (requiring deletion to close rather than
a specific close method), but this generator has gone unused for so long I don't think it would run even
with this patch. Besides, the updated full re-simulation infrastructure allows users to effectively re-sim
from the ECal _perfectly_ so this running mode is already covered.

tomeichlersmith committed:
enable all .py files in test to be tests

the test is just to make sure that they successfully run, nothing is done with the data there, but it is helpful to make sure that basic simulation stuff can run at all during development

tvami committed:
Generalize LHEReader so it can read LHE v3.0 input as well

wesketchum committed:
Merge branch 'wketchum/Genie_Generator' into HEAD

wesketchum committed:
merge in trunk, updates

tomeichlersmith commented:
As discussed on slack, these changes should not be merged before Wednesday and will be [re-applied](https://ldmx-software.github.io/transition-to-ldmx-sw-v4.html) on the subdirectory in ldmx-sw after Wednesday.
